recipe/build.sh

```diff
@@ -23,7 +23,10 @@ for i in `ls`; do
             for j in "$i"/{*.a,*.so,stubs/*.so}*; do
                 # Shared and static libraries are symlinked in $PREFIX/lib
                 ln -sv ${PREFIX}/${targetsDir}/$j ${PREFIX}/$j
-
+                # Provide libcuda.so.1 for applications that need it
+                if [[ $j == "lib/stubs/libcuda.so" ]]; then
```

Comment:
It would be good to have an explicit use case for this symlink. The stub library can't be loaded at runtime and I am worried about this symlink taking priority via `LD_LIBRARY_PATH` etc when a build-system runs an executable that it built.

If this is part of a fix for rpath-link issues with libcuda.so versus libcuda.so.1, we should diagnose what build system is being used. As part of CMake 3.29 we fixed this issue both in the general as we know model ld.bfd requirements, and the specific of libcuda.so

Other build-systems can also model this as direct linking to the libcuda.so stub.

Reply:
Here is the relevant CMake PR: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9050

Reply:
Ok, thanks for the background, yes that's the fix we're addressing. [Here's](https://github.com/conda-forge/cuda-nvcc-feedstock/pull/42) the --rpath-link part of that fix.

So it looks like the changes in both this PR and the other linked above are handled with CMake 3.29 (which is available in conda-forge), is that correct? So both should be closed.

On our side, we've currently got CMake v3.26. We've fixed the issue by [using the libcuda.so.1 symlink from cuda-compat](https://github.com/conda-forge/cuda-compat-feedstock/blob/1d8c74f7d58fe41c5ce974b87ffba02d60d8b13b/recipe/build.sh#L8) and [--rpath-link ing to it](https://github.com/AnacondaRecipes/llama.cpp-feedstock/blob/534d3d39877cf5efe1bdd5cd7c5ae6348fd6df82/recipe/build.sh#L12). This seems to me to be a good solution (the cuda-compat feedstock will be the same version as the cuda-cudart one providing the stubs) although maybe a simpler one would be just making the symlink in a separate directory that's already in --rpath-link?

Reply:
> So it looks like the changes in both this PR and the other linked above are handled with CMake 3.29 (which is available in conda-forge), is that correct? So both should be closed.

That is correct.

> On our side, we've currently got CMake v3.26. We've fixed the issue by [using the libcuda.so.1 symlink from cuda-compat](https://github.com/conda-forge/cuda-compat-feedstock/blob/1d8c74f7d58fe41c5ce974b87ffba02d60d8b13b/recipe/build.sh#L8) and [--rpath-link ing to it](https://github.com/AnacondaRecipes/llama.cpp-feedstock/blob/534d3d39877cf5efe1bdd5cd7c5ae6348fd6df82/recipe/build.sh#L12). This seems to me to be a good solution (the cuda-compat feedstock will be the same version as the cuda-cudart one providing the stubs) although maybe a simpler one would be just making the symlink in a separate directory that's already in --rpath-link?

This solution looks good to me since the `${PREFIX}/cuda-compat/` path is not part of the loader search paths for runtime loading. It fixes the linker issue but wouldn't impact execution of a cuda library, either as part of the build or when installed.